torch/distributed/fsdp/flatten_params_wrapper.py

```diff
@@ -142,6 +144,11 @@ def unflatten_as_params(self) -> Generator:
 
     def __getattr__(self, name: str) -> Any:
         """Forward missing attributes of this wrapper to the wrapped module."""
+        # The `FlatParamHandle`'s existence represents if *this* FPW instance
+        # manages any parameters, so we should not forward the attribute lookup
+        # to nested modules, which also could be FPW instances.
+        if name == FLAT_PARAM_HANDLE:
```

Comment:
Not fully sure if I follow this. Won't this mean that if I have `FSDP(FSDP(model))` then outer FSDP won't have the `.flat_param_handle` attribute, so this lookup would fail?

Reply:
For now, the existence of `_flat_param_handle` is what indicates if the FPW has parameters or not:
https://github.com/pytorch/pytorch/blob/0fd173b097f27b7dd190b25ae13075ba3bf25a5a/torch/distributed/fsdp/flatten_params_wrapper.py#L103-L106
We want the lookup to fail for the outer FSDP.